presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/StatisticsFetcher.java

```diff
@@ -51,18 +52,14 @@ public Stats get()
             return lastStats.get();
         }
 
-        try (InvocationBuilder.AsyncResultCallback<Statistics> callback = new InvocationBuilder.AsyncResultCallback<>()) {
-            DockerClientFactory.lazyClient().statsCmd(container.getContainerId()).exec(callback);
+        CheckedSupplier<Statistics> statisticsSupplier = () -> {
+            try (InvocationBuilder.AsyncResultCallback<Statistics> callback = new InvocationBuilder.AsyncResultCallback<>()) {
+                client.statsCmd(container.getContainerId()).exec(callback);
+                return callback.awaitResult();
+            }
+        };
 
-            return lastStats.getAndUpdate(previousStats -> toStats((Statistics) executor.get(callback::awaitResult), previousStats));
-        }
-        catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
-        catch (RuntimeException e) {
```

Comment:
error handling no longer needed?

Reply:
it's handled by Failsafe now (CheckedSupplier)